src/eth_provider/provider.rs

```diff
@@ -963,15 +963,28 @@ where
             // Extract the signature from the transaction
             let transaction_signature = tx.tx.signature.unwrap();
 
+            // Generate primitive transaction
+            let transaction = rpc_to_primitive_transaction(tx.tx.clone())?;
+
             // Create a signed transaction and send it
             transactions_retried.push(
                 self.send_raw_transaction(
                     TransactionSigned::from_transaction_and_signature(
-                        rpc_to_primitive_transaction(tx.tx.clone())?,
+                        transaction.clone(),
                         reth_primitives::Signature {
                             r: transaction_signature.r,
                             s: transaction_signature.s,
-                            odd_y_parity: transaction_signature.y_parity.unwrap().0,
+                            odd_y_parity: match transaction.tx_type() {
+                                TxType::Legacy => {
+                                    // EIP-155: v = {0, 1} + CHAIN_ID * 2 + 35
+                                    let chain_id = transaction.chain_id().ok_or(TransactionError::InvalidChainId)?;
+                                    let recovery = chain_id.saturating_mul(2).saturating_add(35);
+                                    (transaction_signature.v - U256::from(recovery))
+                                        .try_into()
+                                        .map_err(|_| SignatureError::InvalidParity)?
+                                }
+                                _ => transaction_signature.y_parity.unwrap().0,
+                            },
```

Comment:
to void code duplication, could we use `TransactionEcRecovered::try_from` and extract the signature from there? Or implement a util in `transaction.rs` to retrieve a primitive signature from a rpc signature + chain id

Reply:
Yes this util will be removed from the code after https://github.com/kkrt-labs/kakarot-rpc/pull/1022 but fine to use it here. Just fixed.